components/client/src/main/java/com/hotels/styx/client/StyxHttpClient.java

```diff
@@ -117,18 +116,23 @@ public StreamingTransaction streaming() {
 
         SslContext sslContext = getSslContext(params.https(), params.tlsSettings());
 
-        Mono<LiveHttpResponse> responseObservable = connectionFactory.createConnection(
+        return connectionFactory.createConnection(
                 origin,
                 new ConnectionSettings(params.connectTimeoutMillis()),
                 sslContext
         ).flatMap(connection ->
-                Mono.from(RxReactiveStreams.toPublisher(
-                        connection.write(networkRequest)
-                                .doOnTerminate(connection::close)))
+                Mono.from(connection.write(networkRequest)
```

Comment:
For the record:

As discussed, `Publisher<LiveHttpResponse>` cancel event could occur whilst connection is being used in `flatMap` operator. In this case the connection needs to be closed.

Let's look into this as a separate improvement.